module/VuFind/src/VuFind/Config/Upgrade.php

```diff
@@ -52,83 +54,78 @@ class Upgrade
      *
      * @var array
      */
-    protected $oldConfigs = [];
+    protected array $oldConfigs = [];
 
     /**
      * Processed new configurations
      *
      * @var array
      */
-    protected $newConfigs = [];
+    protected array $newConfigs = [];
 
     /**
      * Comments parsed from configuration files
      *
      * @var array
      */
-    protected $comments = [];
+    protected array $comments = [];
 
     /**
      * Warnings generated during upgrade process
      *
      * @var array
      */
-    protected $warnings = [];
+    protected array $warnings = [];
 
     /**
-     * Are we upgrading files in place rather than creating them?
+     * Have we modified permissions.ini?
      *
      * @var bool
      */
-    protected $inPlaceUpgrade;
+    protected bool $permissionsModified = false;
 
     /**
-     * Have we modified permissions.ini?
+     * If writing of new configuration should be disabled (used in tests).
      *
      * @var bool
      */
-    protected $permissionsModified = false;
+    protected bool $dryRun = false;
 
     /**
      * Constructor
      *
-     * @param string  $from   Version we're upgrading from.
-     * @param string  $to     Version we're upgrading to.
-     * @param string  $oldDir Directory containing old configurations.
-     * @param string  $rawDir Directory containing raw new configurations.
-     * @param ?string $newDir Directory to write updated new configurations into
-     * (leave null to disable writes -- used in test mode).
+     * @param PathResolver  $pathResolver  Path Resolver
+     * @param ConfigManager $configManager Config Manager
      */
     public function __construct(
-        protected string $from,
-        protected string $to,
-        protected string $oldDir,
-        protected string $rawDir,
-        protected ?string $newDir = null
+        protected PathResolver $pathResolver,
+        protected ConfigManager $configManager,
     ) {
-        $this->inPlaceUpgrade = ($this->oldDir == $this->newDir);
     }
 
     /**
      * Run through all of the necessary upgrading.
      *
+     * @param string $newVersion Version to upgrade to
+     * @param bool   $dryRun     If writing of new configuration should be disabled (used in tests)
+     *
      * @return void
      */
-    public function run()
+    public function run(string $newVersion, bool $dryRun = false): void
     {
+        $this->dryRun = $dryRun;
```

Comment:
I generally try to avoid changing class properties as a side effect of other behavior. Would it be better to have dryRun as a testMode constructor parameter rather than making it an argument here? Or would a separate setWriteMode method be a more explicit compromise?